src/include/misc_inline.h

```diff
@@ -315,6 +315,35 @@ __wt_read_shared_double(double *to_read)
     return (*to_read);
 }
 
+/*
+ * __wt_set_shared_maximum64 --
+ *     Given an integer in memory overwrite it with a new value if it is larger.
+ */
+static WT_INLINE void
+__wt_set_shared_maximum64(uint64_t *shared_value, uint64_t new_val)
+{
+    uint64_t current;
+
+    current = *shared_value;
+    while (current < new_val) {
+#if defined(__GNUC__) || defined(__clang__)
+        /*
+         * We can't use the WiredTiger compare and swap as it doesn't allow the second argument to
+         * be a pointer. This method automatically places the value of shared_value into current on
+         * failure.
```

Comment:
This comment is not clear since both #if and #else implementations work.  A better comment would be:
/*
 * This #if is a performance optimization. If *shared_value != current then __atomic_compare_exchange_n() returns the value of *shared_value in current but __wt_atomic_cas64() does not.  Thus __atomic_compare_exchange_n() is more efficient. The #if is needed because Windows does not have __atomic_compare_exchange_n() and must use __wt_atomic_cas64().
 */